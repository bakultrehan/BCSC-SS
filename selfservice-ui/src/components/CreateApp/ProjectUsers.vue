/** * CreateApp of app */

<template>
  <v-card class="v-form pa-8 pt-6 ma-3">
    <v-card-title class="headline padding-0">{{ title }}</v-card-title>
    <v-card-subtitle class="text-left padding-0">Tell us about your {{ title }}</v-card-subtitle>
    <Input
      v-model="userDetails.firstName"
      label="First Name"
      :rules="[rules.required, rules.length(2), rules.maxLength(250)]"
      type="text"
    />
    <Input
      v-model="userDetails.lastName"
      label="Last Name"
      :rules="[rules.required, rules.length(2), rules.maxLength(250)]"
      type="text"
    />
    <Input
      v-model="userDetails.email"
      label="Work email"
      :rules="[rules.required, rules.email, rules.maxLength(250)]"
      type="text"
    />
    <Input
      v-model="userDetails.phone"
      label="Phone"
      :rules="[rules.required, rules.maxLength(15)]"
      type="text"
    />
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';

import { ProjectUserModel, ProjectInfoModel } from '@/models/ProjectInfoModel';
import Input from '@/Atomic/Input/Input.vue';

@Component({
  components: {
    Input
  }
})
export default class ProjectUsers extends Vue {
  private e6: number = 0;
  @Prop({
    default: { email: '', phone: '', firstName: '', lastName: '' }
  })
  private userDetails!: ProjectUserModel;
  @Prop({
    default: ''
  })
  private title!: string;
  @Prop({})
  private rules!: any;
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.padding-0 {
  padding-left: 0px !important;
}
</style>
