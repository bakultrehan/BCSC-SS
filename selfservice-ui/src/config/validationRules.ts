const rules = {
  length: (len: any) => (v: any) =>
    (v || '').length >= len || 'Invalid character length, required ' + len,
  required: (v: any) => !!v || 'This field is required',
  email: (v: any) =>
    /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(v) ||
    'E-mail must be valid'
};

export default rules;
