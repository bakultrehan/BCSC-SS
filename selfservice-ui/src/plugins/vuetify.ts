import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
