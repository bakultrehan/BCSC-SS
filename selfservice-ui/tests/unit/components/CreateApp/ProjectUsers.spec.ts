import { shallowMount, createLocalVue } from '@vue/test-utils';
import vuetify from 'vuetify';
import ProjectUsers from '@/components/CreateApp/ProjectUsers.vue';
import validationRules from '@/config/validationRules';

describe('ProjectUsers.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(ProjectUsers, {
      localVue,
      propsData: {
        userDetails: { email: '', phone: '', firstName: '', lastName: '' },
        rules: validationRules
      }
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
