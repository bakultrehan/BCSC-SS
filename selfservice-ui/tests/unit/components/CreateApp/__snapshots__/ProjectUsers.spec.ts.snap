// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectUsers.vue renders props when passed 1`] = `
<v-card-stub
  class="v-form pa-8 pt-6 ma-3"
  loaderheight="4"
  tag="div"
>
  <v-card-title-stub
    class="headline padding-0"
  >
    
  </v-card-title-stub>
   
  <v-card-subtitle-stub
    class="text-left padding-0"
  >
    Tell us about your 
  </v-card-subtitle-stub>
   
  <input-stub
    counter=""
    label="First Name"
    outlined="true"
    rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
    type="text"
    value=""
  />
   
  <input-stub
    counter=""
    label="Last Name"
    outlined="true"
    rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
    type="text"
    value=""
  />
   
  <input-stub
    counter=""
    label="Work email"
    outlined="true"
    rules="v => !!v || 'This field is required',v => /^\\\\w+([.-]?\\\\w+)*@\\\\w+([.-]?\\\\w+)*(\\\\.\\\\w{2,10})+$/.test(v) || 'E-mail must be valid',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
    type="text"
    value=""
  />
   
  <input-stub
    counter=""
    label="Phone"
    outlined="true"
    rules="v => !!v || 'This field is required',v => (v || '').length <= len || 'Invalid character length, max length ' + len"
    type="text"
    value=""
  />
</v-card-stub>
`;
